File: validParentheses.java
class Solution {
    public boolean isValid(String s) {

        Stack<Character> stack = new Stack<>();
        boolean isTrue = false;

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(' || s.charAt(i) == '[' || s.charAt(i) == '{') {
                stack.push(s.charAt(i));
            }
            else if (s.charAt(i) == ')') {
                if (stack.isEmpty() || stack.peek() != '(') {
                    return false;
                }
                else if (stack.peek() == '(') {
                    stack.pop();
                }
            }
            else if (s.charAt(i) == ']') {
                if (stack.isEmpty() || stack.peek() != '[') {
                    return false;
                }
                else if (stack.peek() == '[') {
                    stack.pop();
                }
            }
            else if (s.charAt(i) == '}') {
                if (stack.isEmpty() || stack.peek() != '{') {
                    return false;
                }
                else if (stack.peek() == '{') {
                    stack.pop();
                }
            }
        }
        if (stack.isEmpty()) {
            isTrue = true;
        }
        else {
            isTrue = false;
        }
        return isTrue;
    }
}

File: Easy/TypeScript/validParentheses.ts
function isValid(s: string): boolean {

    let charArray = Array.from(s);
    let stack = [];
    
    if (charArray.length % 2 !== 0) {
        return false;
    }
    
    for (let i = 0; i < charArray.length; i++) {
        let c = charArray[i];
        if (c === '(' || c === '[' || c === '{') {
            stack.push(c);
        }
        else if (c === ')') {
            if ([...stack].pop() !== '(' || stack.length === 0) {
                return false;
            } else {
                stack.pop();
            }
        }
        else if (c === ']') {
            if ([...stack].pop() !== '[' || stack.length === 0) {
                return false;
            } else {
                stack.pop();
            }
        }
        else if (c === '}') {
            if ([...stack].pop() !== '{' || stack.length === 0) {
                return false;
            } else {
                stack.pop();
            }
        }
    }
    
    if (stack.length === 0) {
        return true;
    }
    
    return false;
};
